src/gui/widgets/settings.cpp

```diff
@@ -84,12 +84,13 @@ static std::vector<std::string>& registered_window_types()
 	return result;
 }
 
-typedef std::map<std::string,
-				 std::function<void(gui_definition&,
-									  const std::string&,
-									  const config&,
-									  const char* key)> >
-tregistered_widget_type;
+struct tregistered_widget_type_mapped_type
+{
+	std::function<styled_widget_definition_ptr(const config&)> parser;
+	const char* key;
+};
+
+typedef std::map<std::string, tregistered_widget_type_mapped_type> tregistered_widget_type;
```

Comment:
that function is already called registered_widget_type and i couldn't really find a good name, will leave this for now.